### ISSUE TITLE
Minting and Burning Specification

### ISSUE THREAD

**Issue by rbharath** [computablelabs/goest#31]
In v2, one of the major changes in `Market.sol` is that market tokens can now be minted and burned by the contract. Contrast with v1 `Registry.sol` which doesn't provide a mechanism for creation of new registry tokens as the registry grows or shrinks.

- Minting: Minting happens in one of three ways.
  - Minting happens when new listings are added to the market. These listings have to pass through the validation process and be whitelisted before minting occurs.
    - `listedReward` is the number of new `MarketTokens` that are created upon whitelisting.
  - Minting happens when an investor enters the market through calling `Market.invest()`. This rate is set by the algorithmic price curve.
  - Minting happens when the backend reports that a listing has been queried. This results in creation of `T_util` new tokens which are awarded to listing owner (which may be the market itself).
- Burning: Burning happens in one of two ways. 
  - If a listing is removed from the market, its associated tokens should be burned. This happens when the listing owner removes the listing or when a successful challenge forces removal of the listing.
  - If an investor class token holder divests from the market, their divested tokens are burned. The origin of the tokens being burned does not matter.

**Comment by robrobbins** [comment]
"whitelisted" has been replaced with "listed".

did we decide that `listReward` is a `Parameterizer` attr ? I think we did

in the event of a deleted listing - the `minted` tokens are burned and the remaining supply, if any, is returned. is this correct?

**Comment by robrobbins** [comment]
`Market.withraw`: a listing owner can withdraw any number of marketTokens <= the `listing.supply`. Should we _not_ allow minted tokens to be withdrawn this way?

**Comment by rbharath** [comment]
Yes, `listReward` is a `Parameterizer` attribute.

Yes, for deleted listings, minted tokens are burned and remaining supply is refunded to listing owner.

For `Market.withdraw`, let's add a new `Parameterizer` attr `mintLockup`. While this lockup period is active, minted tokens cannot be withdrawn by listing owner. After this lockup period passes, minted tokens can be withdrawn, but only if the original listing owner has already surrendered ownership of listing via `Market.convert_to_investor` to the market.

**Comment by robrobbins** [comment]
i think the latter makes the former unnecessary

the simple fact that you cannot withdraw minted tokens unless you are an investor is likely enough

I need to look closer at the `convertToInvestor` flow. 

If I am an investor, this means I have surrendered ownership of a _particular_ listing. Is is true that, as an investor, I may have _other_ listings? This puts me in a sort of "dual state". 

If that is true I then should only be able to withdraw minted tokens from a listing I surrendered. But, is it not true that on the conversion-to-investor for a particular listing All the tokens (the entire supply, including minted) are getting transferred to me anyway?
